File: Leet 2095/solution.go
package main

import (
	"fmt"
)

type ListNode struct {
	Val  int
	Next *ListNode
}

func newNode(value int) *ListNode {
	node := new(ListNode)
	node.Val = value
	return node
}

func deleteMiddle(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return nil
	}

	mid := head
	prev := head
	length := 1
	for curr := head; curr.Next != nil; curr = curr.Next {
		length++

		if length%2 == 0 {
			prev = mid
			mid = mid.Next
		}
	}
	prev.Next = mid.Next
	mid = nil

	return head
}

func createList(nums []int) *ListNode {
	if len(nums) == 0 {
		return nil
	}

	head := newNode(nums[0])
	tail := head
	for i := 1; i < len(nums); i++ {
		tail.Next = newNode(nums[i])
		tail = tail.Next
	}
	return head
}

func printList(head *ListNode) {
	fmt.Print("[")
	for current := head; current != nil; current = current.Next {
		fmt.Printf(" %d ", current.Val)
	}
	fmt.Println("]")
}

func main() {
	fmt.Println("2095. Delete the Middle Node of a Linked List")
	var head *ListNode = createList([]int{1, 3, 4, 7, 1, 2, 6})

	fmt.Println("Before:")
	printList(head)

	fmt.Println("After:")
	head = deleteMiddle(head)
	printList(head)
}

File: Leet 2095/solution.cs
namespace Leet
{
    public class ListNode(int val = 0, ListNode? next = null)
    {
        public int val = val;
        public ListNode? next = next;
    }

    public class Solution
    {
        public ListNode? DeleteMiddle(ListNode? head)
        {
            if (head == null || head.next == null)
            {
                return null;
            }

            var middle = head;
            var prev = head;
            int length = 1;

            for (var current = head; current.next != null; current = current.next)
            {
                // update middle
                if (++length % 2 == 0)
                {
                    prev = middle;
                    middle = middle?.next;
                }
            }

            // remove middle
            prev?.next = middle?.next;

            return head;
        }
    }
}

class Program
{
    private static Leet.ListNode CreateList(int[] nums)
    {
        if (nums.Length == 0)
        {
            return null;
        }

        Leet.ListNode head = new(nums[0]);
        Leet.ListNode tail = head;
        for (int i = 1; i < nums.Length; i++)
        {
            tail.next = new(nums[i]);
            tail = tail.next;
        }
        return head;
    }

    private static void PrintList(Leet.ListNode head)
    {
        Console.Write("[");
        for (var current = head; current != null; current = current.next)
        {
            Console.Write($" {current.val} ");
        }
        Console.WriteLine("]");
    }

    private static void Main()
    {
        Console.WriteLine("2095. Delete the Middle Node of a Linked List");

        int[] nums = [1, 3, 4, 7, 1, 2, 6];
        Leet.Solution solution = new();
        var head = CreateList(nums);

        Console.WriteLine("Before:");
        PrintList(head);
        head = solution.DeleteMiddle(head);
        Console.WriteLine("After:");
        PrintList(head);
    }
}